File: c_language_solutions/11. Container With Most Water.c
int maxArea(int* height, int heightSize) {
    int maxArea = 0;
    int left = 0, right = heightSize - 1;

    while (left < right) {
        int h = fmin(height[left], height[right]);
        int w = right - left;
        int area = h * w;
        maxArea = fmax(maxArea, area);

        if (height[left] < height[right]) {
            left++;
        } else {
            right--;
        }
    }

    return maxArea;
}

File: python_codes/11. Container With Most Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        maxi = 0 
        while left < right:
            gap = min(height[left], height[right]) * (right - left)
            if maxi < gap:
                maxi = gap
            if height[left] <= height[right]:
                left += 1
            else:
                right -= 1
        return maxi
